Content.Server/GameObjects/Components/AI/AiFactionTagComponent.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+
+namespace Content.Server.GameObjects.Components.AI
+{
+    [RegisterComponent]
+    public sealed class AiFactionTagComponent : Component
+    {
+        public override string Name => "AiFactionTag";
+
+        public Faction Factions { get; private set; } = Faction.None;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            var factions = serializer.ReadDataField("factions", new List<Faction>());
```

Comment:
This wouldn't write back to the map any changes made to the factions of this component.
For example, PhysShapeAabb does `serializer.DataField(ref _collisionLayer, "layer", 0, WithFormat.Flags<CollisionLayer>());`

Reply:
WithFormat.Flags<Enum> didn't work content-side last time I tried it with guns, I think comicironic told me not to do it, but I'll see what I can do.